Application/ControlMsrE2/ControlMsrE2Hii.c

```diff
@@ -0,0 +1,397 @@
+                break;
+              case 4:
+                VarStoreValue = *(UINT32 *) (VarPointer);
+                break;
+              default:
+                VarStoreValue = *(UINT64 *) (VarPointer);
+                break;
+              }
+
+              Print (L" Value: value %X", VarStoreValue);
+            }
+            FreePool (Data);
+          }  ///< Allocate
+        }  ///< GetVariable
+        FreePool (VarStoreName);
+        Print (L"\n");
+      }
+    }
+  }
+  FreePool (HiiString);
+}
+
+VOID
+HandleIfrVariable (
+  IN OUT ONE_OF_CONTEXT *Context
+  )
+{
+  EFI_STATUS        Status;
+  EFI_STRING        HiiString;
+  UINTN             DataSize;
+  UINT8             *Data;
+  UINT32            Attributes;
+  UINTN             VarSize;
+  UINT8             *VarPointer;
+  UINT64            VarStoreValue;
+  UINT64            NewValue;
+
+  HiiString = HiiGetString (Context->EfiHandle, Context->IfrOneOf->Question.Header.Prompt, "en-US");
+  if (HiiString) {
+    Print (L"\nBIOS Option found: %s\n", HiiString);
+    FreePool (HiiString);
+  }
+
+  HiiString = AsciiStrCopyToUnicode ((CHAR8 *) Context->IfrVarStore->Name, 0);
+
+  if (HiiString == NULL) {
+    Print (L"\nCould not allocate memory for HiiString\n");
+    return;
+  }
+
+  VarSize = sizeof (EFI_IFR_ONE_OF) - Context->IfrOneOf->Header.Length;
+  VarSize = 8 - (VarSize / 3);
+
+  Print (
+    L"In VarStore \"%s\" GUID: %g Offset: %04X Size: %X ",
+    HiiString,
+    &Context->IfrVarStore->Guid,
+    Context->IfrOneOf->Question.VarStoreInfo.VarOffset,
+    VarSize
+    );
+
+  DataSize = 0;
+
+  Status = gRT->GetVariable (
+                  HiiString,
+                  (VOID *) &Context->IfrVarStore->Guid,
+                  &Attributes,
+                  &DataSize,
+                  NULL
+                  );
+
+  if (Status == EFI_BUFFER_TOO_SMALL) {
+    Data = AllocatePool (DataSize);
+    if (Data != NULL) {
+      Status = gRT->GetVariable (
+                      HiiString,
+                      (VOID *) &Context->IfrVarStore->Guid,
+                      &Attributes,
+                      &DataSize,
+                      Data
+                      );
+
+      if (!EFI_ERROR (Status)) {
+        VarPointer = Data + Context->IfrOneOf->Question.VarStoreInfo.VarOffset;
+        switch (VarSize) {
+        case 1:
+          VarStoreValue = *VarPointer;
+          break;
+        case 2:
+          VarStoreValue = *(UINT16 *) (VarPointer);
+          break;
+        case 4:
+          VarStoreValue = *(UINT32 *) (VarPointer);
+          break;
+        default:
+          VarStoreValue = *(UINT64 *) (VarPointer);
+          break;
+        }
+
+        Print (L"Value: value %X\n", VarStoreValue);
+
+        if (IS_INTERACTIVE ()) {
+          NewValue = VarStoreValue != 0 ? 0 : 1;
+        } else if (IS_LOCK ()) {
+          NewValue = 1;
+        } else if (IS_UNLOCK ()) {
+          NewValue = 0;
+        } else {
```

Comment:
If more than one flags are set in `InterpretArguments`, it will just stop there with `EFI_INVALID_PARAMETER`. And I agree there should be some changes to those modes.

Reply:
I can sort of live with the defines, but I cannot really live them accessing a variable with a very generic name (Flags) without even mentioning it in the arguments. Please rename it to something like mArgumentFlags or something like that.

Reply:
And to be honest, the way it is done, cannot it just be a set of equality checks?